src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/translation.html.twig

```diff
@@ -37,124 +37,161 @@
+                <span>{{ error_count }}</span>
+            </span>
+        {% endif %}
+    </span>
 {% endblock %}
 
 {% block panel %}
     {% if collector.messages is empty %}
         <h2>Translations</h2>
-        <p>
-            <em>No translations have been called.</em>
-        </p>
+        <div class="empty">
+            <p>No translations have been called.</p>
+        </div>
     {% else %}
         {{ block('panelContent') }}
     {% endif %}
 {% endblock %}
 
 {% block panelContent %}
-    {% set filter = request.query.get('state', '-1') %}
-    {% set filterOptions = {
-        '-1': '',
-        (constant('Symfony\\Component\\Translation\\DataCollectorTranslator::MESSAGE_DEFINED')): 'Defined messages',
-        (constant('Symfony\\Component\\Translation\\DataCollectorTranslator::MESSAGE_MISSING')): 'Missing messages',
-        (constant('Symfony\\Component\\Translation\\DataCollectorTranslator::MESSAGE_EQUALS_FALLBACK')): 'Fallback messages',
-    } %}
-
-    <h2>Translation Stats</h2>
-    <table>
-        <tbody>
-            <tr>
-                <th>Defined messages</th>
-                <td><pre>{{ collector.countdefines }}</pre></td>
-            </tr>
-            <tr>
-                <th scope="col" style="width: 30%">Fallback messages</th>
-                <td scope="col" style="width: 60%"><pre>{{ collector.countFallbacks }}</pre></td>
-            </tr>
-            <tr>
-                <th>Missing messages</th>
-                <td><pre>{{ collector.countMissings }}</pre></td>
-            </tr>
+    <h2>Translation Metrics</h2>
+
+    <div class="metrics">
+        <div class="metric">
+            <span class="value">{{ collector.countdefines }}</span>
+            <span class="label">Defined messages</span>
+        </div>
+
+        <div class="metric">
+            <span class="value">{{ collector.countFallbacks }}</span>
+            <span class="label">Fallback messages</span>
+        </div>
+
+        <div class="metric">
+            <span class="value">{{ collector.countMissings }}</span>
+            <span class="label">Missing messages</span>
+        </div>
+    </div>
+
+    <h2>Translation Messages</h2>
+
+    {# sort translation messages in groups #}
+    {% set messages_defined, messages_missing, messages_fallback = [], [], [] %}
+    {% for message in collector.messages %}
+        {% if message.state == constant('Symfony\\Component\\Translation\\DataCollectorTranslator::MESSAGE_DEFINED') %}
+            {% set messages_defined = messages_defined|merge([message]) %}
+        {% elseif message.state == constant('Symfony\\Component\\Translation\\DataCollectorTranslator::MESSAGE_MISSING') %}
+            {% set messages_missing = messages_missing|merge([message]) %}
+        {% elseif message.state == constant('Symfony\\Component\\Translation\\DataCollectorTranslator::MESSAGE_EQUALS_FALLBACK') %}
+            {% set messages_fallback = messages_fallback|merge([message]) %}
+        {% endif %}
```

Comment:
You are right again :)  But if we add a new state, I'm afraid that it's not going to be displayed in any way. Sorting messages is only part of the solution. The other part is actually displaying those messages and this is not ready for other states. But we can easily deal with this when the time comes.